src/Workspaces/Core/MSBuild.BuildHost/RemoteProjectFile.cs

```diff
@@ -29,4 +29,61 @@ public Task<ImmutableArray<ProjectFileInfo>> GetProjectFileInfosAsync(Cancellati
     public Task<ImmutableArray<DiagnosticLogItem>> GetDiagnosticLogItemsAsync(CancellationToken cancellationToken)
         => Task.FromResult(_projectFile.Log.ToImmutableArray());
 
+    public Task<string> GetDocumentExtensionAsync(SourceCodeKind sourceCodeKind, CancellationToken cancellationToken)
+        => Task.FromResult(_projectFile.GetDocumentExtension(sourceCodeKind));
+
+    public Task AddDocumentAsync(string filePath, string? logicalPath, CancellationToken cancellationToken)
```

Comment:
all of these methods complete synchronously.  consider making ValueTasks

Reply:
File no longer exists.